theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
> we might want to check other frameworks' implementations again (linked at #5621 (comment)). If there's no clear argument for either layout, we might opt for compatibility (to make it easier to port models between frameworks).

Regarding this, [torch](https://github.com/soumith/cudnn.torch/blob/f746b79/SpatialConvolution.lua#L22) and [caffe](https://github.com/BVLC/caffe/blob/bbc4e57/src/caffe/layers/base_conv_layer.cpp#L137) have used the same memory layout for the weights, I have used the same layout in the new PR as well.
Tensorflow's layout is quite different [here ](https://www.tensorflow.org/versions/r0.11/api_docs/python/nn/convolution#depthwise_conv2d),we probably cannot port to tensorflow
So, shall I continue with the method being implemented in the new PR?